src/pages/iou/ReceiptSelector/NavigationAwareCamera.js

```diff
@@ -0,0 +1,51 @@
+import React, {useEffect, useState} from 'react';
+import {Camera} from 'react-native-vision-camera';
+import {useTabAnimation} from '@react-navigation/material-top-tabs';
+import {useNavigation} from '@react-navigation/native';
+import refPropTypes from '../../../components/refPropTypes';
+
+const propTypes = {
+    /* Forwarded ref */
+    forwardedRef: refPropTypes.isRequired,
+};
+
+// Wraps a camera that will only be active when the tab is focused or as asoon as it start to become focused.
+function NavigationAwareCamera(props) {
+    // Get navigation to get initial isFocused value (only needed once during init!)
+    const navigation = useNavigation();
+    const [isCameraActive, setIsCameraActive] = useState(navigation.isFocused);
+    // Get the animation value from the tab navigator. Its a value between 0 and the
+    // number of pages we render in the tab navigator. When we even just slightly start
+    // to scroll to the camera page, we want to activate the camera (so its as fast as possible active).
+    const tabPositionAnimation = useTabAnimation();
+
+    useEffect(() => {
+        const listenerId = tabPositionAnimation.addListener(({value}) => {
+            setIsCameraActive(value > 0 && value < 2);
```

Comment:
Not sure if this is possible, but can we dynamically get the number of tabs so it's less brittle?

Reply:
Yes, let me update the code, so its dynamic, good call!

Reply:
I made it now dynamic, wdyt?